### PR TITLE
add service extract_embedded  & url analysis & OTX lookup and fix bug in others services

### PR DESCRIPTION
**Pull request by lprat**
Hi,
I created a new service (extract embedded), which to use clamscan to extract embedded file as new sample in crits.
I hope that service will interest you.

Thank you for your tool & work !

Regards
Lionel

### CODE REVIEW

File: extract_embedded_service/__init__.py [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+            obj.filedata.seek(0)
+            #make temp dir for extract embedded file
+            dtmp = tempfile.mkdtemp()
+            #create empty file for no check sig on file
+            emptyrule_path = tempfile.gettempdir() + '/emptyrule.yar'
+            if not os.path.isfile(emptyrule_path):
+                open(emptyrule_path, 'a').close()
+            #TODO add filename filter for avoid escape shell
+            args = [clamscan_path, '--quiet', '--leave-temps', '--tempdir=' + dtmp, '-d', emptyrule_path, filename]
+            if not user.has_access_to(RawDataACL.WRITE):
```
Comment (mgoffin) [inline extract_embedded_service/__init__.py]: Does this service write anything to RawData? If not, does this prevent the user from using the service if they don't have RawData Write permissions even though it doesn't interact with RawData?

File: extract_embedded_service/__init__.py [hunk 2]
```diff
@@ -0,0 +1,148 @@
...
+                self._info("User does not have permission to add Raw Data to CRITs")
+                self._add_result("Parsing Cancelled", "User does not have permission to add Raw Data to CRITs")
+                return
+            # pdftotext does not generate a lot of output, so we should not have to
```
Comment (mgoffin) [inline extract_embedded_service/__init__.py]: A copy error left over from the service you mirrored this one on?

**Comment by lprat** [comment]
Hi,
I do error in my code with RawData, i use sampleACL! I changed it and i'm added condition for user not autorised to write, who can see result without write sample.
I'm not sure to have understood your second question. But i modified the code in order that if one error happen, the code remove all temporary files excepted yara file empty.

Regards, 
Lionel

File: extract_embedded_service/__init__.py (hunk 2 above)
Comment (lprat) [inline extract_embedded_service/__init__.py]: I'm not sure to have understood your question. But i modified the code in order that if one error happen, the code remove all temporary files excepted yara file empty.

File: extract_embedded_service/__init__.py (hunk 1 above)
Comment (lprat) [inline extract_embedded_service/__init__.py]: I do error in my code with RawData, i use sampleACL! I changed it and i'm added condition for user not autorised to write, who can see result without write sample.

**Comment by lprat** [comment]
I added corection for service extract_macro and pdf2file.
I had error when run command: "Error running service: Cannot encode object: <SimpleLazyObject: <CRITsUser: lionel>>".
When remove field 'analyst' to call function "obj.add_relationship", that error disappear.
I changed  RelationshipTypes RELATED_TO by CONTAINED_WITHIN, but crits display relationtype "Contains" instead of "contained within"...

**Comment by mgoffin** [comment]
It seems you are adding a lot more code to this PR than was intended. I would suggest breaking this PR up into it separate ones for each feature/enhancement/bugfix you are attempting to submit.

**Comment by lprat** [comment]
I create separate pull request with news services and bug fix.

**Pull request closed without merging**